metadata-ingestion/src/datahub/ingestion/api/auto_work_units/auto_ensure_aspect_size.py

```diff
@@ -81,6 +95,195 @@ def ensure_schema_metadata_size(
 
         schema.fields = accepted_fields
 
+    def ensure_query_subjects_size(
```

Comment:
Could we make this method a bit "smarter" and explicitly avoid adding schema fields for subjects which we dropped at table level? Would we actually ever hit this case? Or should we assume (that's my opinion) that if we start trimming on table level subjects, then for sure we don't have enough space to fine-grained subjects? This seems logical for me, maybe we could already return from function when hitting the limitation for the first time in line 127?, similarly, maybe it doesn't make sense to continue iterating over fine-grained subjects once we already hit the limitation - assuming URNs are somehow similar in size, it wouldn't make much sense.

Reply:
But we are still iterating through all the fine-grained subjects, even if we already started trimming on table level, am I wrong?

Reply:
addressed in 59d9e3d7bd5b717319d02626d010b6677c298689